test/test_train_mp_mnist_amp.py

```diff
@@ -130,14 +130,21 @@ def train_mnist(flags, **kwargs):
   lr = flags.lr * xm.xrt_world_size()
 
   device = xm.xla_device()
+  device_hw = xm.xla_device_hw(device)
   model = MNIST().to(device)
   writer = None
   if xm.is_master_ordinal():
     writer = test_utils.get_summary_writer(flags.logdir)
   optim_cls = syncfree.SGD if FLAGS.use_syncfree_optim else optim.SGD
   optimizer = optim_cls(model.parameters(), lr=lr, momentum=flags.momentum)
   loss_fn = nn.NLLLoss()
-  scaler = GradScaler(use_zero_grad=FLAGS.use_zero_grad)
+  if device_hw == 'TPU':
+        autocast = torch.xla.amp.autocast
```

Comment:
Question: can you point me to the mapping the explains what op is `autocast`ed and what op is not?

Reply:
The op list is registered on the pytorch side here: https://github.com/pytorch/pytorch/blob/8455aac198e4ee68003f2b38a5e15631fb82690c/aten/src/ATen/autocast_mode.cpp#L520